src/main/kotlin/com/hibob/academy/employeeFeedback/service/FeedbackService.kt

```diff
@@ -0,0 +1,70 @@
+package com.hibob.academy.employeeFeedback.service
+
+import com.hibob.academy.employeeFeedback.dao.*
+import jakarta.ws.rs.BadRequestException
+import org.springframework.stereotype.Service
+
+@Service
+class FeedbackService(private val feedbackDao: FeedbackDao) {
+    fun feedbackLengthValidation(feedback: String): Boolean {
+        val minCharacters = 30
+        val maxCharacters = 500
+        return !(feedback.length < minCharacters || feedback.length > maxCharacters)
+    }
+
+    fun anonymityStatusValidation(employeeId: Int?): Int{
+        return employeeId ?: throw Exception("Can't check status of anonymous feedback")
+    }
+
+    fun authorizationToViewValidation(employeeId: Int, feedbacksEmployeeId: Int) {
+        if (feedbacksEmployeeId != employeeId)
+            throw BadRequestException("Unauthorized to view this feedback status")
+    }
+
+    fun submitFeedback(employeeId: Int, companyId: Int, anonymity: AnonymityType, feedback: String): Boolean {
+        var employeeIdForFeedback: Int? = employeeId
+
+        if (anonymity == AnonymityType.ANONYMOUS)
+            employeeIdForFeedback = null
+
+        val validation = feedbackLengthValidation(feedback)
+
+        if (validation) {
```

Comment:
you can do it nicer:

if (!validation) return false

and then the rest of the code, not in a if. Then the code looks better